lib/virtus/support/type_lookup.rb

```diff
@@ -84,7 +82,7 @@ def determine_type_from_primitive(primitive)
       descendants.select(&:primitive).reverse_each do |descendant|
         descendant_primitive = descendant.primitive
         next unless primitive <= descendant_primitive
-        type = descendant if type.nil? or type.primitive > descendant_primitive
+        type = descendant if type.nil? || type.primitive > descendant_primitive
```

Comment:
What do you mean? It is correct now, isn't it? `||` and `&&` should be only used in conditions (like above)